libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayEventSimulator.kt

```diff
@@ -16,9 +18,12 @@ import kotlinx.coroutines.launch
 /**
  * This class keeps track of a forward playing replay. As time moves forward, it captures
  * all events from [ReplayEvents] that happened, and provides them in a [ReplayEvents]
+ *
+ * @param logger interface for logging any events
  */
 internal class ReplayEventSimulator(
-    private val replayEvents: ReplayEvents
+    private val replayEvents: ReplayEvents,
+    private val logger: Logger?
```

Comment:
Should we add a default value here (`null`) 🤔? We're adding it to `ReplayHistoryMapper`